services/web/server/src/simcore_service_webserver/meta_api_handlers_repos.py

```diff
@@ -0,0 +1,323 @@
+import logging
+from typing import List
+
+from aiohttp import web
+from pydantic.decorator import validate_arguments
+from servicelib.rest_pagination_utils import PageResponseLimitOffset
+
+from ._meta import api_version_prefix as vtag
+from .login.decorators import login_required
+from .meta_api_handlers_base import (
+    create_url_for_function,
+    enveloped_response,
+    handle_request_errors,
+)
+from .meta_core_repos import (
+    checkout_checkpoint_safe,
+    create_checkpoint_safe,
+    get_checkpoint_safe,
+    get_workbench_safe,
+    list_checkpoints_safe,
+    list_repos_safe,
+    update_checkpoint_safe,
+)
+from .meta_db import HEAD, VersionControlRepository
+from .meta_models_repos import (
+    Checkpoint,
+    CheckpointAnnotations,
+    CheckpointApiModel,
+    CheckpointNew,
+    RefID,
+    RepoApiModel,
+    WorkbenchView,
+    WorkbenchViewApiModel,
+)
+from .rest_utils import RESPONSE_MODEL_POLICY
+from .security_decorators import permission_required
+from .utils_aiohttp import rename_routes_as_handler_function, view_routes
+
+logger = logging.getLogger(__name__)
+
+
+# FIXME: access rights using same approach as in access_layer.py in storage.
+# A user can only check snapshots (subresource) of its project (parent resource)
+
+
+@validate_arguments
+def _normalize_refid(ref_id: RefID) -> RefID:
+    if ref_id == "HEAD":
+        return HEAD
+    return ref_id
+
+
+# API ROUTES HANDLERS ---------------------------------------------------------
+routes = web.RouteTableDef()
+
+
+@routes.get(f"/{vtag}/repos/projects")
+@login_required
+@permission_required("project.read")
+@handle_request_errors
+async def _list_repos_handler(request: web.Request):
+    # FIXME: check access to non owned projects user_id = request[RQT_USERID_KEY]
+    url_for = create_url_for_function(request)
+    vc_repo = VersionControlRepository(request)
+
+    _limit = int(request.query.get("limit", 20))
```

Comment:
should we not have a shared constant DEFAULT_ITEMS_PER_PAGE across the entire app?
@sanderegg I don't remember if you already added something like this.

Reply:
will add in ``packages/service-library/src/servicelib/rest_pagination_utils.py``